data/styles/vehicle/include/Basemap.mapcss

```diff
@@ -526,48 +522,88 @@ area|z17-[amenity=parking_space]
 
 /* 8.BUILDINGS */
 
-area|z12-[man_made=pier],
-area|z12-[man_made=breakwater],
-line|z14-[man_made=pier],
-line|z14-[man_made=breakwater],
-line|z14-[man_made=cutline]
-{fill-position: background;	fill-opacity: 1;}
+/* 8.1 Pier, dam, cutline */
 
-/* 8.1 Pier */
-area|z12-[man_made=pier],
-area|z12-[man_made=breakwater],
- {fill-color: @background;casing-color: @background;}
+area|z13-[man_made=pier],
+area|z13-[man_made=breakwater],
+{fill-position: background; fill-opacity: 0.7; fill-color: @background;casing-color: @background;}
 
-line|z14-[man_made=pier],
-line|z14-[man_made=breakwater],
-line|z14-[man_made=cutline]
-{casing-width: 2;casing-color: @background;fill-color: @background;}
+line|z15-[man_made=pier],
+line|z15-[man_made=breakwater],
+{width: 2;color: @background;opacity: 0.7;}
+
+line|z15-[waterway=dam],
+line|z15-[waterway=weir],
+{width: 2;color: @bridge_casing;opacity: 0.7;}
+
+line|z16[man_made=pier],
+line|z16[man_made=breakwater],
+line|z16[man_made=cutline],
+line|z16[waterway=dam],
+line|z16[waterway=weir],
+{width: 3;}
+
+line|z17[man_made=pier],
+line|z17[man_made=breakwater],
+line|z17[man_made=cutline],
+line|z17[waterway=dam],
+line|z17[waterway=weir],
+{width: 4.5;}
+
+line|z18[man_made=pier],
+line|z18[man_made=breakwater],
+line|z18[man_made=cutline],
+line|z18[waterway=dam],
+line|z18[waterway=weir],
+{width: 6;}
+
+line|z19-[man_made=pier],
+line|z19-[man_made=breakwater],
+line|z19-[man_made=cutline],
+line|z19-[waterway=dam],
+line|z19-[waterway=weir],
+{width: 8;}
+
+line|z15[man_made=cutline],
+{width: 1.5;color: @green0;opacity: 0.5;}
+line|z16[man_made=cutline],
+{width: 2;}
+line|z17[man_made=cutline],
+{width: 3;}
+line|z18-[man_made=cutline],
+{width: 4.5;color: @green1;}
+line|z19-[man_made=cutline],
+{width: 6;}
 
 /* 8.2 Building */
 
 
 area|z15[building],
+area|z15[waterway=dam],
 area|z15[landuse=garages],
 area|z15[building=train_station],
 {fill-color: @building;fill-opacity: 0.65;casing-width: 1;casing-color: @building_border;}
 
 area|z14-15[aeroway=terminal],
 area|z16[building],
 area|z16[building:part],
+area|z16[waterway=dam],
 area|z16[landuse=garages],
 area|z16[building=train_station],
 {fill-color: @building;fill-opacity: 0.55;casing-width: 1;casing-color: @building_border;}
 
 area|z16[aeroway=terminal],
 area|z17[building],
 area|z17[building:part],
+area|z17[waterway=dam],
 area|z17[landuse=garages],
 area|z17[building=train_station],
 {fill-color: @building;opacity: 0.55;casing-width: 1;casing-color: @building_border;}
 
 area|z17-[aeroway=terminal],
 area|z18-[building],
 area|z18-[building:part],
+area|z18-[waterway=dam],
 area|z18-[landuse=garages],
 area|z18-[building=train_station]
 {fill-color: @building;opacity: 0.55;casing-width: 1;casing-color: @building_border;}
```

Comment:
Can we combine last 3 blocks? They have equal drawing rules.

Reply:
And I suspect that last 2 blocks should define fill-opacity. Combine 3 blocks into one.